wallet/service.go

```diff
@@ -150,6 +152,44 @@ func (service *Service) LinkBitFlyerWallet(ctx context.Context, info *walletutil
 	return nil
 }
 
+// LinkGeminiWallet links a wallet and transfers funds to newly linked wallet
+func (service *Service) LinkGeminiWallet(ctx context.Context, info *walletutils.Info, verificationToken string) error {
+	// get gemini client from context
+	geminiClient, ok := ctx.Value(appctx.GeminiClientCTXKey).(gemini.Client)
+	if !ok {
+		// no gemini client on context
+		return handlers.WrapError(
+			appctx.ErrNotInContext, "gemini client misconfigured", http.StatusInternalServerError)
+	}
+
+	// perform an Account Validation call to gemini to get the accountID
+	accountID, err := geminiClient.ValidateAccount(ctx, verificationToken)
+	if err != nil {
+		return handlers.WrapError(
+			errors.New("invalid linking_info"), "unable to validate gemini account", http.StatusBadRequest)
+	}
+
+	// we assume that since we got linking_info(VerificationToken) signed from Gemini that they are KYC
+	providerLinkingID := uuid.NewV5(WalletClaimNamespace, accountID)
+	if info.ProviderLinkingID != nil {
+		// check if the member matches the associated member
+		if !uuid.Equal(*info.ProviderLinkingID, providerLinkingID) {
+			return handlers.WrapError(errors.New("wallets do not match"), "unable to match wallets", http.StatusForbidden)
+		}
+	} else {
+		// tx.Destination will be stored as UserDepositDestination in the wallet info upon linking
+		err := service.Datastore.LinkWallet(ctx, info.ID, accountID, providerLinkingID, nil, "gemini")
```

Comment:
is `return fmt.Errorf("error trying to lookup anonymous address: %w", err)` in `TxLinkWalletInfo(` worth changing the status for?

Reply:
server error seems appropriate for now.